### PR TITLE
Add warning for upcoming Python default change

### PR DESCRIPTION
**Pull request by BanzaiMan**
To correspond to #1678.

### CODE REVIEW

File: lib/travis/build/script/python.rb [hunk 1]
```diff
@@ -7,6 +7,15 @@ class Python < Script
           virtualenv: { system_site_packages: false }
         }
 
+        DEPRECATIONS = [
+          {
+            name: 'Python',
+            current_default: DEFAULTS[:phython],
```
Comment (cotsog) [inline lib/travis/build/script/python.rb]: Is this a typo?

```suggestion
            current_default: DEFAULTS[:python],
```

Comment (emdantrim) [inline lib/travis/build/script/python.rb]: `phython`

**Comment by BanzaiMan** [comment]
> Quick one, where's the code using the `DEPRECATIONS` array?

It's here: https://github.com/travis-ci/travis-build/blob/a8d57ff7ca73f2d5dcef93f2ac61bbece4010465/lib/travis/build/script.rb#L445-L456

**Comment by cotsog** [comment]
Thanks! 🐑:it:

File: lib/travis/build/script/python.rb (hunk 1 above)
Comment (BanzaiMan) [reply lib/travis/build/script/python.rb]: 🙈

**Pull request merged**